src/graphics/engine/text.cpp

```diff
@@ -261,6 +261,8 @@ void CText::FlushCache()
     m_lastCachedFont = nullptr;
     m_lastFontType = FONT_COMMON;
     m_lastFontSize = 0;
+
+    Create();
```

Comment:
Again, you are calling Create() multiple times - this is bad approach and most likely does not free all resources of the previous instance